### PR TITLE
Make label checks more dynamic

### LINKED ISSUE #22: Changelog workflow doesn't skip "late" label
## Problem Summary
Changes that do not warrant a changelog issue still fail if the label is added after the PR creation, until the next push

## Impact
The checks can fail in situations where they shouldn't

## Possible Solution
One option is to trigger on label changes, but there may be better methods.
Until fixed, whitespace commits will fully re-evaluate the checks

### PR DESCRIPTION
**Pull request by diehlbw**
# Overview

Closes #22 

## Description of changes
Use gh api to request labels dynamically instead of using the cached list that requires an event (commit) to renew. 
Split changelog workflow into two jobs; could not find a method for short-circuiting success within a job.  

Note: this method is intentionally requiring manual retry, and NOT triggering off of label changes

Suggested testing-  
As a step 0, I'm intentionally NOT setting labels prior to PR creation.  
Tests can add and remove labels on this PR, then manually rerun the specific workflow.  
Verify that the second job runs and is skipped depending on the skip-changelog label.  
Before merge, update labels and drop the second commit to verify the push event also behaves correctly

## Author Checklist
- [ ] Linting passes; run early with [pre-commit hook](https://pre-commit.com/#install).
- [ ] Tests added for new code and issue being fixed.
- [ ] Added type annotations and full numpy-style docstrings for new methods.
- [ ] Draft your news fragment in new `changelog/ISSUE.TYPE.rst` files; see [changelog/README.md](https://github.com/epic-open-source/seismometer/blob/main/changelog/README.md).

### CODE REVIEW

**Comment by diehlbw** [comment]
Initial run 'successfully' set needs_entry.  
Rerun after setting a different label, timed out on first attempt(?) then set it correctly.

File: .github/workflows/changelog.yml [hunk 1]
```diff
@@ -6,11 +6,32 @@ on:
...
+            echo "needs_entry=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "needs_entry=false" >> "$GITHUB_OUTPUT"
+          fi
+    outputs:
+      needs_entry: ${{ steps.check-labels.outputs.needs_entry }}
+
   check-changelog-entry:
-    name: changelog entry
+    name: has fragment
```
Comment (firste) [inline .github/workflows/changelog.yml:31]: Can we make the name clearer for this? Like "has changelog fragment"?

File: .github/workflows/changelog.yml [hunk 2]
```diff
@@ -6,11 +6,32 @@ on:
 permissions: "read-all"
```
Comment (firste) [inline .github/workflows/changelog.yml:6]: I think we only need "pull-requests: read" permission here.

**Review by firste — CHANGES_REQUESTED** [review]

File: .github/workflows/changelog.yml (hunk 2 above)
Comment (diehlbw) [reply .github/workflows/changelog.yml]: FINC

File: .github/workflows/changelog.yml (hunk 1 above)
Comment (diehlbw) [reply .github/workflows/changelog.yml]: works for me

**Review by firste — APPROVED** [review]

**Pull request merged**